src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java

```diff
@@ -1376,18 +1377,17 @@ private void expressionBasedAttack(String param, String modifiedParamValue, Stri
 		countExpressionBasedRequests++;
 
 		String modifiedExpressionOutputUnstripped = msg.getResponseBody().toString();
-		String modifiedExpressionOutputStripped = this.stripOff(modifiedExpressionOutputUnstripped, modifiedParamValue);
+		String modifiedExpressionOutputStripped = stripOffOriginalAndAttackParam(modifiedExpressionOutputUnstripped, originalParam, modifiedParamValue);
 
-		//set up little arrays to ease the work of checking the unstripped output, and then the stripped output
-		String normalBodyOutput[] = {mResBodyNormalUnstripped, mResBodyNormalStripped};
-		String expressionBodyOutput[] = {modifiedExpressionOutputUnstripped, modifiedExpressionOutputStripped};
-		boolean strippedOutput[] = {false, true};
+		//give the variable a nicer name
+		String normalBodyOutput = mResBodyNormalStripped;
```

Comment:
If the variable name isn't useful for development/maintenance use then just replace it throughout. No point wasting memory storing the same string twice or CPU cycles assigning it twice.